test/test_torch.py

```diff
@@ -12699,6 +12699,85 @@ def test_fill_diagonal(self):
         e1.fill_diagonal_(v, wrap=True)
         self.assertEqual(e1, e2)
 
+    def test_binary_op_input_output_overlap(self):
+        sz = 3
+        data = torch.randn(3 * sz)
+
+        # c is identical to a and b:
+        c = data[0:sz]
+        c_exp = c.clone()
+        a = data[0:sz]
+        a_exp = a.clone()
+        b = data[0:sz]
+        b_exp = b.clone()
+        torch.add(a_exp, b_exp, out=c_exp)
```

Comment:
You can just write `c_exp = a + b` to avoid all the extra cloning here and below.